executor/builder.go

```diff
@@ -1236,15 +1236,74 @@ func (b *executorBuilder) buildUpdate(v *plan.Update) Executor {
 		b.err = errors.Trace(b.err)
 		return nil
 	}
+	columns2Handle := buildColumns2Handle(v.Schema(), tblID2table)
 	updateExec := &UpdateExec{
-		baseExecutor: newBaseExecutor(b.ctx, nil, v.ExplainID(), selExec),
-		SelectExec:   selExec,
-		OrderedList:  v.OrderedList,
-		tblID2table:  tblID2table,
+		baseExecutor:   newBaseExecutor(b.ctx, nil, v.ExplainID(), selExec),
+		SelectExec:     selExec,
+		OrderedList:    v.OrderedList,
+		tblID2table:    tblID2table,
+		columns2Handle: columns2Handle,
 	}
 	return updateExec
 }
 
+// cols2Handle represents an mapper from column index to handle index.
+type cols2Handle struct {
+	start, end    int32 // Represent the ordinal range [start, end) of the consecutive columns.
+	handleOrdinal int32 // Represents the ordinal of the handle column.
+}
+
+// cols2HandleSlice attaches the methods of sort.Interface to []cols2Handle sorting in increasing order.
+type cols2HandleSlice []cols2Handle
+
+// Len implements sort.Interface#Len.
+func (c cols2HandleSlice) Len() int {
+	return len(c)
+}
+
+// Len implements sort.Interface#Swap.
+func (c cols2HandleSlice) Swap(i, j int) {
+	c[i], c[j] = c[j], c[i]
+}
+
+// Len implements sort.Interface#Less.
+func (c cols2HandleSlice) Less(i, j int) bool {
+	return c[i].start < c[j].start
+}
+
+// findHandle finds the ordinal of the corresponding handle column.
+func (c cols2HandleSlice) findHandle(ordinal int32) (int32, bool) {
+	if c == nil || len(c) == 0 {
+		return 0, false
+	}
+	biggerOne := sort.Search(len(c), func(i int) bool { return c[i].start > ordinal })
+	if biggerOne == 0 {
+		return 0, false
+	}
+	if ordinal < c[biggerOne-1].end {
+		return c[biggerOne-1].handleOrdinal, true
+	}
+	return 0, false
+}
+
+// buildColumns2Handle build columns to handle mapping.
+func buildColumns2Handle(schema *expression.Schema, tblID2Table map[int64]table.Table) cols2HandleSlice {
+	if len(schema.TblID2Handle) < 2 {
```

Comment:
1. We need a comment for this check.
2. Any test case covers this?

Reply:
old case tested this...for `update single table update t set x =1 where y = 1`

...but I had added one in executor_test TestUpdateJoin~